lib/getters/cartGetters.ts

```diff
@@ -87,25 +88,64 @@ export const getDiscounts = (cart: Cart) => {
   }))
 }
 
-export const getCartFulfillmentOptions = (item: CartItem, purchaseLocation: Location) =>
-  item?.product.fulfillmentTypesSupported.map((option) => ({
-    name: "fulfillment",
-    value: option,
-    label: option === "DirectShip" ? "Ship to Home" : "Pickup in Store",
-    details:
-      option === "DirectShip"
-        ? "Available to Ship"
-        : purchaseLocation?.name
-        ? `Available at: ${purchaseLocation.name}`
+const isDisabledFulfillmentOption = (fulfillmentTypesSupported, option) => {
+  return !fulfillmentTypesSupported?.includes(option)
+}
+
+const getFulfillmentDetails = (option, item, cartItemFulfillmentLocation: string) => {
+  const details = {
+    DirectShip: option.details,
+    InStorePickup:
+      item?.fulfillmentMethod === "Pickup" && cartItemFulfillmentLocation
+        ? `Available at ${cartItemFulfillmentLocation}`
         : "",
-    required: "false",
+  }
+
+  return isDisabledFulfillmentOption(item.product?.fulfillmentTypesSupported, option.value)
+    ? "Not Available"
+    : "" || details[option.value]
+}
+
+const getFulfillmentDescription = (cartItemFulfillmentLocation: string) => {
+  if (cartItemFulfillmentLocation) {
+    return "Change Store"
```

Comment:
Using this hardcoded strings inside getters isn't best practice. 
Should be in the component and should be i18n locale translation

Reply:
Done